File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        n = len(nums)
        res = [0 for _ in range(n)]
        
        left, right = 0, n - 1
        index = n - 1
        while left <= right:                
            if abs(nums[left]) <= abs(nums[right]):
                res[index] = nums[right] ** 2
                right -= 1
            else:
                res[index] = nums[left] ** 2
                left += 1
            index -= 1
        return res
                
                

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        vector<int> result(nums.size());
        int left = 0;
        int right = nums.size() - 1;
        int k = nums.size() - 1;
        for (; k>= 0; k--){
            if (nums[right] * nums[right] >= nums[left] * nums[left]){
                result[k] = nums[right] * nums[right];
                right--;
            }
            else{
                result[k] = nums[left] * nums[left];
                left++;
            }
        }
        return result;
        
    }
};